rpcserver.go

```diff
@@ -679,13 +679,50 @@ func (r *rpcServer) Start() error {
 
 		go func() {
 			rpcsLog.Infof("gRPC proxy started at %s", lis.Addr())
-			http.Serve(lis, mux)
+			if len(cfg.RestCORS) == 0 {
+				http.Serve(lis, mux)
+			} else {
+				http.Serve(lis, checkCors(mux))
+			}
 		}()
 	}
 
 	return nil
 }
 
+// checkCors allows Cross Origin Resoruce Sharing from configured origins.
+func checkCors(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origin := r.Header.Get("Origin")
+
+		// Skip CORS check if request did not send origin
+		if origin != "" {
+			// Check if all origins are allowed or requested origin is in list
+			for _, allowedOrigin := range cfg.RestCORS {
+				if allowedOrigin == "*" || origin == allowedOrigin {
+					// Only set ACAO header for requested origin
+					w.Header().Set("Access-Control-Allow-Origin", origin)
+
+					// Set necessary headers for preflight requests
+					if r.Method == "OPTIONS" &&
+						r.Header.Get("Access-Control-Request-Method") != "" {
+						w.Header().Set("Access-Control-Allow-Headers",
```

Comment:
nit: add new line between condition and code block

Reply:
Good catch, that was looking a little weird.